src/main/java/ch/njol/skript/expressions/ExprExperience.java

```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 	
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		double d;
-		if (e instanceof ExperienceSpawnEvent)
-			d = ((ExperienceSpawnEvent) e).getSpawnedXP();
-		else if (e instanceof BlockBreakEvent)
-			d = ((BlockBreakEvent) e).getExpToDrop();
-		else
+		double eventExp;
+		if (e instanceof ExperienceSpawnEvent) {
+			eventExp = ((ExperienceSpawnEvent) e).getSpawnedXP();
+		} else if (e instanceof BlockBreakEvent) {
+			eventExp = ((BlockBreakEvent) e).getExpToDrop();
+		} else if (e instanceof PlayerExpChangeEvent) {
+			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+		} else {
 			return;
-		
-		if (delta != null)
```

Comment:
This check should be restored (likely at the beginning of the method) to avoid any useless work (e.g. setting the exp amount to the same number when delta is null)